### PR TITLE
Feature/registering domains

### PR DESCRIPTION
**Pull request by danielcarney96**
# Subdomain registration

**Ticket:** N/A

## Description

This PR adds the initial subdomain registration form as part of the registration process.

The user flow for registration can be found in `docs/flowcharts/user_registration.md`, instructions for viewing charts can be found in `docs/charts.md`.

The subdomain registration step is found after registration a fresh account without an invite, doing so will then redirect the user to their new domain. For example, creating a subdomain named 'something' will then direct the user to 'something.localhost'.

The next step will be to define some permissions, probably using guards to separate subdomain and project permissions. Then we can limit access to subdomains as its currently open.

## Checklist

- [x] The `develop` branch has been merged into this branch and conflicts resolved
- [x] All existing tests are passing
- [x] Manual testing has been completed
- [x] Automated tests have been added where necessary
- [x] Documentation has been updated where necessary

### CODE REVIEW

File: database/seeders/DatabaseSeeder.php [hunk 1]
```diff
@@ -13,6 +14,11 @@ class DatabaseSeeder extends Seeder
      */
     public function run()
     {
-        // \App\Models\User::factory(10)->create();
+        User::factory()->create([
+            'first_name' => 'Daniel',
```
Comment (danbfield) [inline database/seeders/DatabaseSeeder.php:18]: Rude, where is mine?

File: resources/js/Pages/Auth/Subdomain.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+export interface ISubdomainProps {}
+
+export const Subdomain: React.FC<ISubdomainProps> = (props) => {
+  const { data, setData, post, processing, errors, reset } = useForm({
+    name: '',
+  })
+
+  const onHandleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setData(event.target.name as 'name', event.target.value)
```
Comment (danbfield) [inline resources/js/Pages/Auth/Subdomain.tsx]: You could de-structure `event.target` like: `const { name, value } = event.target`.

File: database/seeders/DatabaseSeeder.php (hunk 1 above)
Comment (danielcarney96) [reply database/seeders/DatabaseSeeder.php:18]: Make it yourself you lazy bum

File: resources/js/Pages/Auth/Subdomain.tsx (hunk 2 above)
Comment (danielcarney96) [reply resources/js/Pages/Auth/Subdomain.tsx]: Good shout

**Pull request merged**